code/datums/status_effects/buffs.dm

```diff
@@ -454,3 +454,50 @@
 	name = "Nest Vitalization"
 	desc = "The resin seems to pulsate around you. It seems to be sustaining your vital functions. You feel ill..."
 	icon_state = "nest_life"
+
+/**
+ * Granted to wizards upon satisfying the cheese sacrifice during grand rituals.
+ * Halves incoming damage and makes the owner stun immune, damage slow immune, levitating(even in space and hyperspace!) and glowing.
+ */
+/datum/status_effect/blessing_of_insanity
+	id = "blessing_of_insanity"
+	duration = -1
+	tick_interval = -1
+	alert_type = /atom/movable/screen/alert/status_effect/blessing_of_insanity
+
+/atom/movable/screen/alert/status_effect/blessing_of_insanity
+	name = "Blessing of Insanity"
+	desc = "Your devotion to madness has improved your resilience to all damage and you gain the power to levitate!"
+	//no screen alert - the gravity already throws one
+
+/datum/status_effect/blessing_of_insanity/on_apply()
+	if(ishuman(owner))
+		var/mob/living/carbon/human/human_owner = owner
+		human_owner.physiology.brute_mod *= 0.5
+		human_owner.physiology.burn_mod *= 0.5
+		human_owner.physiology.tox_mod *= 0.5
+		human_owner.physiology.oxy_mod *= 0.5
+		human_owner.physiology.clone_mod *= 0.5
+		human_owner.physiology.stamina_mod *= 0.5
```

Comment:
cache the `physiology` var to a typed var like `var/datum/physiology/editable = human_owner.physiology` for cleanliness and whatever speed that might yield

Reply:
i don't know the actual type of the physiology datum but you know what i meant. typecast cache stuffs